client/src/components/sidebar/SidebarRecentFileItem.vue

```diff
@@ -0,0 +1,88 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ion-item
+    lines="none"
+    button
+    @click="$emit('fileClicked', file)"
+    class="sidebar-item menu-default"
+    ref="itemRef"
+  >
+    <ion-label class="sidebar-item-file-label">{{ file.name }}</ion-label>
+    <div
+      class="file-option"
+      @click.stop="$emit('removeClicked', file)"
+    >
+      <ion-icon :icon="close" />
+    </div>
+  </ion-item>
+</template>
+
+<script setup lang="ts">
+import { IonItem, IonLabel, IonIcon } from '@ionic/vue';
+import { close } from 'ionicons/icons';
+import { RecentFile } from '@/services/recentFiles';
+
+defineProps<{
+  file: RecentFile;
+}>();
+
+defineEmits<{
+  (e: 'fileClicked', file: RecentFile): void;
+  (e: 'removeClicked', file: RecentFile): void;
+}>();
+</script>
+
+<style scoped lang="scss">
+.sidebar-item {
+  --background: none;
+  border-radius: var(--parsec-radius-8);
+  border: solid 1px var(--parsec-color-light-primary-800);
+  --min-height: 0;
+
+  .file-option {
+    color: var(--parsec-color-light-secondary-grey);
+    text-align: right;
+    position: absolute;
+    display: flex;
+    align-items: center;
+    top: 0;
+    right: 1rem;
+    font-size: 1.2rem;
+    padding-top: 0.5rem;
+    opacity: 0;
+
+    &:hover {
+      color: var(--parsec-color-light-primary-30);
+    }
+  }
+
+  &:hover {
+    border: solid 1px var(--parsec-color-light-primary-30-opacity15);
+    cursor: pointer;
+
+    .file-option {
+      opacity: 1;
+    }
+  }
+
+  &:active,
+  &.item-selected {
+    --background: var(--parsec-color-light-primary-30-opacity15);
+  }
+
+  & > ion-label {
+    --color: var(--parsec-color-light-primary-100);
+  }
+
+  &-file-label {
+    position: relative;
+    margin-right: 1.1rem;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+    --color: var(--parsec-color-light-primary-100);
+  }
+}
```

Comment:
This style block seems to be a duplicate from existing parts of the sidebar menu.

It should be factorized by common rules or with a common component for sidebar subparts.